### ISSUE TITLE
why detach(stop-gradient) the boxes for PrPooling

### ISSUE THREAD

**Issue by GMonster0905** [VISION-SJTU/USOT#1]
The contribution of precise ROI-pooling can be concluded as: pooling more precise ROI features and can backpropagate the gradients to the coordinates. 

We find you detach the boxes when pooling the results in the training pipeline at 
https://github.com/VISION-SJTU/USOT/blob/main/lib/models/models.py#L273.
that's somewhat confusing for using cycle consistency to train an unsupervised tracking model.  If dont detach these boxes, the model may acquire more info from the video-level training and can be better trained ? But now this part is detached, the model seems like the conventional pair-wise trained siamese tracking model. So can we ask, why this part is detached ? what if this part is not detached?

**Comment by zhengjilai** [comment]
Thanks for your question.

The basic inference logic of bbox regression based Siamese tracker is that, only one output bbox is finally used for outputting the final bounding box. That is the box on the regression map with the highest score on the classification map. The main reason for our design of bbox detachment is that, we think it is not proper to pass gradient from only one spatial position of the regression map. Note that when training the regression map like Siamese network, the gradients are passed backward from all bboxes whose positions are inside the gt bbox, so not detaching cycle gradient for bboxes may cause a misalignment in this aspect. 

One fact I may share is that, cycle consistency can be trained with either stop-gradient manner or not-stop-gradient manner. The former is the implementation philosophy of UDT, and the latter is the paradigm of TimeCycle. In TimeCycle , gradient is passed back both from the output transformation and the feature. You may refer to these papers for details. 

I really do not know what would happen if that detachment is removed. However, I really guess that depending on gradient from only a single position on the regression map is not enough.
